src/consensus/pool.rs

```diff
@@ -238,11 +237,32 @@ impl PoolImpl {
         certs
     }
 
-    /// Fetches all votes cast by myself for any slots starting from `slot`.
-    fn get_own_votes(&self, slot: Slot) -> Vec<Vote> {
+    /// Fetches finalization certficates for given `slot`.
+    ///
+    /// Prefers fast-finalization over slow-finalization, if it's available.
+    /// In that case this returns only the fast-finalization certificate.
+    /// Otherwise, returns the finalization and notarization certificates.
+    fn get_final_certs(&self, slot: Slot) -> Vec<Cert> {
+        let Some(slot_state) = self.slot_states.get(&slot) else {
+            return Vec::new();
+        };
+
+        if let Some(ff_cert) = slot_state.certificates.fast_finalize.clone() {
+            vec![Cert::FastFinal(ff_cert)]
+        } else if let Some(final_cert) = slot_state.certificates.finalize.clone()
+            && let Some(notar_cert) = slot_state.certificates.notar.clone()
+        {
+            vec![Cert::Final(final_cert), Cert::Notar(notar_cert)]
+        } else {
+            vec![]
+        }
+    }
```

Comment:
```suggestion
    /// Fetches finalization certficates for given `slot`.
    ///
    /// Prefers fast-finalization over slow-finalization, if it's available.
    /// In that case this returns only the fast-finalization certificate.
    /// Otherwise, returns the finalization and notarization certificates.
    fn get_final_certs(&self, slot: Slot) -> Vec<Cert> {
        let Some(slot_state) = self.slot_states.get(&slot) else {
            return Vec::new();
        };
        if let Some(ff_cert) = &slot_state.certificates.fast_finalize {
            return vec![Cert::FastFinal(ff_cert.clone())];
        }
        if let Some(final_cert) = &slot_state.certificates.finalize
            && let Some(notar_cert) = &slot_state.certificates.notar
        {
            return vec![
                Cert::Final(final_cert.clone()),
                Cert::Notar(notar_cert.clone()),
            ];
        }
        vec![]
    }

```

I have a small preference for early returns instead of using `else`.  No worries if you do not like that style.  The other change is to clone later to save some potentially unnecessarily clones.